src/llm/mod.rs

```diff
@@ -48,6 +51,48 @@ pub trait LLM: Sync + Send {
     async fn generate(&self, prompt: Box<dyn Prompt>) -> Result<LLMResponse>;
 }
 
+/// Embedding trait is used to generate an embedding from an Online Service.
+#[async_trait::async_trait]
+pub trait EMBEDDING: Sync + Send {
```

Comment:
Also nit: This can be `Embedding` since it is not an acronym

Reply:
Easy, I wasn't sure on the best naming scheme.